### ISSUE TITLE
Stemming override ability

### ISSUE THREAD

**Issue by tommiv** [imotov/elasticsearch-analysis-morphology#18]
Is it possible to make stemmer exceptions? For example we need to not stem word "Мечел" as it's company name. I've tried to follow [this guide](http://www.elastic.co/guide/en/elasticsearch/guide/current/controlling-stemming.html) with no luck.

```
"analysis": {
            "analyzer": {
                "ru": {
                    "type": "custom",
                    "tokenizer": "standard",
                    "filter": ["lowercase", "custom_stem", "russian_morphology", "english_morphology", "ru_stopwords"]
                }
            },
            "filter": {
                "ru_stopwords": {
                    "type": "stop",
                    "stopwords": "<...skipped...>"
                },
                "no_stem": {
                    "type": "keyword_marker",
                    "keywords": ["мечел"]
                },
                "custom_stem": {
                    "type": "stemmer_override",
                    "rules": [ 
                        "мечел=>мечел",
                        "мечеть=>мечеть",
                    ]
                }
            }
        }
```

Neither `no_stem` or `custom_stem` are working, I always get "мечел" => "мечеть" stem result. 

```
curl -XGET 'http://localhost:12001/posts/_analyze?analyzer=ru' -d 'мечел'
{"tokens":[{"token":"мечеть","start_offset":0,"end_offset":5,"type":"<ALPHANUM>","position":1}]}
```

Update: `"stem_exclusion": ["мечел"],` inside `analyzer=>ru` also doesn't work.

**Comment by imotov** [comment]
I have opened PR in the upstream project to add support for keywords  AKuznetsov/russianmorphology#1

**Comment by tommiv** [comment]
This is great news, thanks!

**Comment by comdiv** [comment]
Highly required. And I don't know how to fix it myself while it problem of underlying MorphologyFilter of Lucene itself. Cannot prevent stemming of townname "Богданович" from normalizing to "богдан" with keyword_repeat filter.

**Comment by comdiv** [comment]
Have fix it in my fork of stable 1.2.1 while i need hot fix on production. Simply copy MorphologyFilter of LUCENE, fix it and reset factories on it. You can see it in https://github.com/comdiv/elasticsearch-analysis-morphology/commit/975f8f33f5dfddaf5b122fba6c2021bddc3b278e
